pennylane/ops/qubit.py

```diff
@@ -2051,6 +2082,113 @@ def __init__(self, control_wires=None, wires=None, control_values=None, do_queue
+
+            if len(self._work_wires) >= num_work_wires_needed:
+                decomp = self._decomposition_with_many_workers(
+                    self.control_wires, self._target_wire, self._work_wires
+                )
+            else:
+                work_wire = self._work_wires[0]
+                decomp = self._decomposition_with_one_worker(
+                    self.control_wires, self._target_wire, work_wire
+                )
+
+        flips2 = [
+            qml.PauliX(self.control_wires[i])
+            for i, val in enumerate(self.control_values)
+            if val == "0"
+        ]
+
+        return flips1 + decomp + flips2
+
+    @staticmethod
+    def _decomposition_with_many_workers(control_wires, target_wire, work_wires):
+        """Decomposes the multi-controlled PauliX gate using the approach in Lemma 7.2 of
+        https://arxiv.org/pdf/quant-ph/9503016.pdf, which requires a suitably large register of
+        work wires"""
+        num_work_wires_needed = len(control_wires) - 2
+        work_wires = work_wires[:num_work_wires_needed]
+
+        work_wires_reversed = list(reversed(work_wires))
+        control_wires_reversed = list(reversed(control_wires))
+
+        gates = []
+
+        for i in range(len(work_wires)):
+            ctrl1 = control_wires_reversed[i]
+            ctrl2 = work_wires_reversed[i]
+            t = target_wire if i == 0 else work_wires_reversed[i - 1]
+            gates.append(qml.Toffoli(wires=[ctrl1, ctrl2, t]))
+
+        gates.append(qml.Toffoli(wires=[*control_wires[:2], work_wires[0]]))
+
+        for i in reversed(range(len(work_wires))):
+            ctrl1 = control_wires_reversed[i]
+            ctrl2 = work_wires_reversed[i]
+            t = target_wire if i == 0 else work_wires_reversed[i - 1]
+            gates.append(qml.Toffoli(wires=[ctrl1, ctrl2, t]))
+
+        for i in range(len(work_wires) - 1):
+            ctrl1 = control_wires_reversed[i + 1]
+            ctrl2 = work_wires_reversed[i + 1]
+            t = work_wires_reversed[i]
+            gates.append(qml.Toffoli(wires=[ctrl1, ctrl2, t]))
+
+        gates.append(qml.Toffoli(wires=[*control_wires[:2], work_wires[0]]))
+
+        for i in reversed(range(len(work_wires) - 1)):
+            ctrl1 = control_wires_reversed[i + 1]
+            ctrl2 = work_wires_reversed[i + 1]
+            t = work_wires_reversed[i]
+            gates.append(qml.Toffoli(wires=[ctrl1, ctrl2, t]))
+
+        return gates
+
+    @staticmethod
+    def _decomposition_with_one_worker(control_wires, target_wire, work_wire):
+        """Decomposes the multi-controlled PauliX gate using the approach in Lemma 7.3 of
+        https://arxiv.org/pdf/quant-ph/9503016.pdf, which requires a single work wire"""
+        tot_wires = len(control_wires) + 2
+        partition = int(np.ceil(tot_wires / 2))
+
+        first_part = control_wires[:partition]
+        second_part = control_wires[partition:]
+
+        gates = [
+            MultiControlledX(
```

Comment:
They do need to be decomposed, but the n-controlled operation is broken down into 4 smaller m-controlled operations such that there are always enough unused qubits that they can be used as work qubits. Like in the example:
![image](https://user-images.githubusercontent.com/2068515/117717715-28fa1700-b1a9-11eb-98a0-5e77a58153b6.png)
With the extra work qubit there, the 7-controlled gate turns into a 5-controlled gate and there are conveniently 3 leftover qubits that could be used as scratch space.

That said, yes, there is a tradeoff because you can tinker with the relative sizes of the four smaller `MultiControlledX` gates. I think the optimum is ceil(n / 2) (which is what gets used here).